src/main.rs

```diff
@@ -225,6 +228,12 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
         }
         None => (),
     }
+    if matches.opt_present(json_option) {
+        let smbios_data = table_load_from_device()?;
```

Comment:
So, to get at the defined structure information and make it look more interesting than a bunch of bytes, `table_load_from_device` returns an `SMBiosData`.  Within `SMBiosData` we have:
1. a table
2. a version

The table is an `UndefinedStructTable` and this is why it prints bytes out instead of parsing it pretty.  We have to get from UndefinedStructTable to DefinedStructTable.

The impl Debug for UndefinedStructTable says this is how you do it:
```rust
        // Convert to defined structures to see the structure fields
        let defined_table: DefinedStructTable<'_> = self.table.iter().collect();
```
What enables the above statement to work is `collect()` and that trick for implementing a collect() ability a `FromIterator` trait in `DefinedStructTable`:

```rust
impl<'a> FromIterator<&'a UndefinedStruct> for DefinedStructTable<'a> {
    fn from_iter<I: IntoIterator<Item = &'a UndefinedStruct>>(iter: I) -> Self {
        let mut defined_struct_table = DefinedStructTable::new();

        for undefined_struct in iter {
            defined_struct_table.add(undefined_struct.into());
        }

        defined_struct_table
    }
}
```

1. The first problem we should probably solve is producing JSON that represents the `DefinedStructTable`, since it is the most useful to people.
2. The second problem is doing something with Version information and I have an idea that will take a little explaining.  The version information comes from the Entry Point.  But there are two entry point structures (32 and 64 bit) plus on Windows you can get the information without having either of those structures.  It would be good to be able to give the original structure it came from.  However, the most important thing that people will want to know is the version.  I'll share the design I'm thinking in a moment.